test_runner/src/main/kotlin/ftl/api/Artifacts.kt

```diff
@@ -0,0 +1,34 @@
+package ftl.api
+
+import ftl.adapter.GcFetchArtifacts
+
+val fetchArtifacts: Artifacts.Fetch get() = GcFetchArtifacts
+
+object Artifacts {
+
+    data class Identity(
+        val gcsPathWithoutRootBucket: String,
+        val gcsRootBucket: String,
+        val regex: List<Regex>,
+        val blobPath: String, // todo not needed?
+        val downloadPath: DownloadPath,
+        val matrixId: String
+    )
+
+    data class DownloadPath(
+        val localResultDir: String,
+        val useLocalResultDir: Boolean,
+        val keepFilePath: Boolean
+    )
+
+    /**
+     * Fetches all files for provided [Identity].
+     * @return Matrix ID of which files were downloaded
+     * associated with list of downloaded files
+     */
+    interface Fetch {
+        suspend operator fun invoke(input: Identity): Pair<MatrixId, List<String>>
+    }
+}
+
+inline class MatrixId(val value: String)
```

Comment:
Inline classes as an experiment, let me know what do you think. I can change it to simple string ofc.

@jan-gogo

Reply:
@pawelpasterz besides the idea of a performance increase(which i am unsure is necessary in our case) what benefit do they bring?

Reply:
@Sloox Intention is rather code robustness. The inline class brings more information on what a primitive (string in that case) represents. It will be unwrapped to `String` (during compilation) so I think there is no benefit in performance.

It's definitely not something that I need to add, I'm just curious about your opinion.

Reply:
Inline classes could be useful for further issues when it will come to wrapping direct structured output.

But for this case, there is no need for a dedicated type for matrix id, because there is no type checking on it. The `private typealias` is sufficient.